UI/system-tray-icon.cpp

```diff
@@ -0,0 +1,298 @@
+#include "system-tray-icon.hpp"
+#include "window-basic-main.hpp"
+
+#include <QAction>
+
+OBSSystemTrayIcon::OBSSystemTrayIcon(QWidget *parent) : QSystemTrayIcon(parent)
+{
+#ifdef __APPLE__
+	QIcon trayIconFile = QIcon(":/res/images/obs_macos.svg");
+	trayIconFile.setIsMask(true);
+#else
+	QIcon trayIconFile = QIcon(":/res/images/obs.png");
+#endif
+	setIcon(QIcon::fromTheme("obs-tray", trayIconFile));
+	setToolTip("OBS Studio");
+
+	OBSBasic *main = OBSBasic::Get();
```

Comment:
Rather than doing this, why not pass it as a constructor parameter ?

Reply:
Fixed